scripts/build/builders/bouffalolab.py

```diff
@@ -153,20 +205,41 @@ def __init__(self,
             self.argsOpt.append('chip_enable_additional_data_advertising=true')
             self.argsOpt.append('chip_enable_rotating_device_id=true')
 
+        if "disable" != function_mfd:
```

Comment:
how about using an enum for mfd options?  Free-form strings are harder to ensure that they meet available support.

Reply:
change to enum